lib/features/Auth/login/cubit/login_methods.dart

```diff
@@ -0,0 +1,177 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:mentorship_e1_g3/core/networking/auth_exception.dart';
+import 'package:mentorship_e1_g3/core/routing/app_routing.dart';
+import 'package:mentorship_e1_g3/features/Auth/login/cubit/login_cubit.dart';
+import 'package:mentorship_e1_g3/features/home/presentation/screen/home_screen.dart';
+
+import '../../../../core/helpers/functions/snakbar.dart';
+import '../presentation/widgets/otp_bottomsheet.dart';
+
+abstract interface class LoginMethods {
+  login(BuildContext context);
+}
+
+class GoogleLogin implements LoginMethods {
```

Comment:
In this file there is many functions, consider refactoring them by making smaller function of each one.